python/cudf/cudf/core/multiindex.py

```diff
@@ -414,7 +417,13 @@ def __eq__(self, other):
             return False
         # Lazy comparison
         if isinstance(other, MultiIndex) or hasattr(other, "_source_data"):
-            return self._source_data.equals(other._source_data)
+            for self_col, other_col in zip(
+                self._source_data._data.values(),
+                other._source_data._data.values(),
+            ):
+                if not self_col.equals(other_col):
+                    return False
+            return self.names == other.names
```

Comment:
Why can't we just do DataFrame level equality checking here?

Reply:
Connected offline and the reasoning is that MultiIndexes can have the names `[None, None, None, ...]` which isn't supported in a DataFrame and the equality check would fail on the names.